src/MessagePack.UnityClient/Assets/Scripts/Tests/ShareTests/ExpandoObjectTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) All contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Dynamic;
+using System.Runtime.Serialization;
+using MessagePack.Resolvers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MessagePack.Tests
+{
+    public class ExpandoObjectTests
+    {
+        private readonly ITestOutputHelper logger;
+
+        public ExpandoObjectTests(ITestOutputHelper logger)
+        {
+            this.logger = logger;
+        }
+
+        [Fact]
+        public void ExpandoObject()
+        {
+            var options = MessagePackSerializerOptions.Standard;
+
+            dynamic expando = new ExpandoObject();
+            expando.Name = "George";
+            expando.Age = 18;
+
+            byte[] bin = MessagePackSerializer.Serialize(expando, options);
+            this.logger.WriteLine(MessagePackSerializer.ConvertToJson(bin));
+
+            dynamic expando2 = MessagePackSerializer.Deserialize<ExpandoObject>(bin, options);
+            Assert.Equal(expando.Name, expando2.Name);
+            Assert.Equal(expando.Age, expando2.Age);
+        }
+
+        [Fact]
+        public void ExpandoObject_DeepGraphContainsAnonymousType()
+        {
+            var options = MessagePackSerializerOptions.Standard;
+            var deserializerOptions = MessagePackSerializerOptions.Standard.WithResolver(
+                CompositeResolver.Create(
+                    PrimitiveObjectResolver.InstanceWithExpandoObject,
+                    MessagePackSerializerOptions.Standard.Resolver));
```

Comment:
This is somewhat cumbersome. I'd at least like to have just one set of resolvers used both for serializing and deserializing. And if we could reasonably expose a pre-composed resolver, so much the better. But I'm not sure how to do that, nor how worthwhile the effort is.